pennylane/pauli/grouping/group_observables.py

```diff
@@ -119,75 +152,326 @@ def binary_repr(self, n_qubits=None, wire_map=None):
+                else (itemgetter(*indices)(items),)
+            )
+            for indices in idx_partitions
+        )
+    else:
+        items_partitioned = [
+            (
+                list(itemgetter(*indices)(items))
+                if len(indices) > 1
+                else [itemgetter(*indices)(items)]
+            )
+            for indices in idx_partitions
         ]
 
-        return self.grouped_paulis
+    return items_partitioned
+
+
+def _adj_matrix_from_symplectic(symplectic_matrix: np.ndarray, grouping_type: str):
+    """Get adjacency matrix of (anti-)commuting graph based on grouping type.
+
+    This is the adjacency matrix of the complement graph. Based on symplectic representations and inner product of [1].
+
+    [1] Andrew Jena (2019). Partitioning Pauli Operators: in Theory and in Practice.
+    UWSpace. http://hdl.handle.net/10012/15017
+
+    Args:
+        symplectic_matrix (np.ndarray): 2D symplectic matrix. Each row corresponds to the
+        symplectic representation of the Pauli observables.
+        grouping_type (str): the binary relation used to define partitions of
+            the Pauli words, can be ``'qwc'`` (qubit-wise commuting), ``'commuting'``, or
+            ``'anticommuting'``.
+
+    Returns:
+        np.ndarray: Adjacency matrix. Binary matrix such that adj_matrix[i,j] = 1 if observables[i]
+        observables[j] do **not** (anti-)commute, as determined by the ``grouping_type``.
+    """
+
+    n_qubits = symplectic_matrix.shape[1] // 2
+
+    # Convert symplectic representation to integer format.
+    # This is equivalent to the map: {0: I, 1: X, 2:Y, Z:3}
+    pauli_matrix_int = 2 * symplectic_matrix[:, :n_qubits] + symplectic_matrix[:, n_qubits:]
+    pauli_matrix_int = pauli_matrix_int.astype(np.int8)
+    # Broadcast the second dimension, sucht that pauli_matrix_broad.shape = (m, 1, n_qubits)
+    # with m = len(observables). This allows for calculation of all possible combinations of Pauli observable pairs (Pi, Pj).
+    # Something like: result[i, j, k] = pauli_matrix_int[i, k] * pauli_matrix_int[j, k]
+    pauli_matrix_broad = pauli_matrix_int[:, None]
+    # Calculate the symplectic inner product in [1], using the integer representation - hence the difference in the equation form.
+    # qubit_anticommutation_mat[i,j, k] is k=0 if Pi and Pj commute, else k!=0 if they anticommute.
+    qubit_anticommutation_mat = (pauli_matrix_int * pauli_matrix_broad) * (
+        pauli_matrix_int - pauli_matrix_broad
+    )
+    # 'adjacency_mat[i, j]' is True iff Paulis 'i' and 'j' do not (anti-)commute under given grouping_type.
+    if grouping_type == "qwc":
+        # True if any term anti commutes
+        adj_matrix = np.logical_or.reduce(qubit_anticommutation_mat, axis=2)
+    elif grouping_type == "commuting":
+        # True if the number of anti commuting terms is odd (anti commte)
+        adj_matrix = np.logical_xor.reduce(qubit_anticommutation_mat, axis=2)
```

Comment:
Am I missing something, or are these two cases exactly the same?

Reply:
QWC is using the `or` and the commuting is using `xor` :).  This is because the `or` would not be satisfied (true in this case) as soon as one of the term does not commute, while the `xor` would actually care about the parity of terms not satisfying the binary relation

Reply:
well now I see it...